src/plugins/data/public/ui/query_string_input/query_string_input.test.tsx

```diff
@@ -94,16 +100,17 @@ describe('QueryStringInput', () => {
     jest.clearAllMocks();
   });
 
-  it('Should render the given query', () => {
-    const component = mount(
+  it('Should render the given query', async () => {
+    const { getByText } = render(
       wrapQueryStringInputInContext({
         query: kqlQuery,
         onSubmit: noop,
         indexPatterns: [stubIndexPatternWithFields],
       })
     );
-    expect(component.find(EuiTextArea).props().value).toBe(kqlQuery.query);
-    expect(component.find(QueryLanguageSwitcher).prop('language')).toBe(kqlQuery.language);
+
+    await waitFor(() => getByText(kqlQuery.query));
+    await waitFor(() => getByText('KQL'));
```

Comment:
`waitFor` already fails if the items you are querying for don't exist, so we no longer need the assertions that were here previously.